proto/lambda.proto

```diff
@@ -0,0 +1,195 @@
+message Type {
+    message Unit { }
+    message Int { }
+    message Int32 { }
+    message Int64 { }
+    message Bool { }
+    message String { }
+    message Lwt { }
```

Comment:
How to handle values of `Lwt` type? E.g. [some block operations](https://github.com/mirage/mirage-lambda/blob/4922cd4ecbdb30f69904fac366c931e06a947932/mirage/unikernel.ml#L26-L34) have `lwt` return types. Is the value serialization not implemented, yet, or am I overlooking something?

Reply:
You need to use `Apply` and `Lwt` (like `Apply (Int, Lwt)`). This about a limitation of the OCaml type-system (and the higher kind polymorphism). We can provide a sugar about that if you want.

Reply:
Okay, thanks. Just to verify, would the following be a correct encoding of the program
```
λ (output_block : page_aligned_buffer).
  Block.read 42 output_block
```
(using the definitions from [here](https://github.com/mirage/mirage-lambda/blob/4922cd4ecbdb30f69904fac366c931e06a947932/mirage/unikernel.ml#L12-L35))

```
lam {
  typ { abstract { witness: "Cstruct.t" } }
  var: "output_block"
  expr {
    app {
      a {
        app {
          a {
            prm {
              value {
                name: "Block.read"
                arguments { int64 { } }
                arguments { list { value { abstract { witness: "Cstruct.t" } } } }
                return {
                  apply {
                    a {
                      result { a { unit { } } b { abstract { witness: "Block.error" } } }
                    }
                    b { lwt { } }
                  }
                }
                smartptr: 4
              }
            }
          }
          b { val { value { int64 { value: 42 } } } }
        }
      }
      b { var { var: 0 } }
    }
  }
}
```

Reply:
Exactly 👍 , as I said, we can provide a sugar and translate `lwt { something }` to `apply { a { something } b { lwt { } }` internally if it's better for you.